ihaskell-display/ihaskell-widgets/src/IHaskell/Display/Widgets/Button.hs

```diff
@@ -58,7 +61,17 @@ mkButton = do
 instance IHaskellDisplay Button where
   display b = do
     widgetSendView b
-    return $ Display []
+    return $ Display
+      [ DisplayData
+          (MimeCustom "application/vnd.jupyter.widget-view+json")
+          (L.toStrict
+            (encodeToLazyText
+              (object
+                [ "model_id" .= uuid b
+                , "version_major" .= (2 :: Integer)
+                , "version_minor" .= (0 :: Integer)
+                ])))
+      ]
```

Comment:
I'm battling a little bit with getting the order of messages correct with this approach. It appears as though:

```
b <- mkButton
b
```

works ok.

However, for:

```
mkButton
```

the `display_data` message arrives before `comm_open` does.

Do you possibly have a hint as to why? My instinct was that `flushWidgetMessages` would need to preceed https://github.com/gibiansky/IHaskell/blob/c50e5fc15ff7dcf784fc06dcd3f7318b525f8c98/main/Main.hs#L194, but I'm not having much success with that.

Reply:
I don't have much insight into how `ihaskell-widgets` works, so I unfortunately can't be of much help here.

Reply:
No worries, thanks

Reply:
solved with https://github.com/gibiansky/IHaskell/pull/1091/commits/27981f9b20c1562307b1742cc41b7976d1b7545e